apiserver/facades/controller/sshtunneler/facade.go

```diff
@@ -45,3 +49,18 @@ func (f *Facade) RemoveSSHConnRequest(arg params.SSHConnRequestRemoveArg) (param
 	}
 	return params.ErrorResult{}, nil
 }
+
+// ControllerAddresses returns the specified machine's public addresses.
```

Comment:
a conceptual question: do we want machine agents to ssh to public addresses or private ones? idk which is guaranteed to be reachable from the unit

Reply:
@wallyworld @hpidcock any thoughts on this?

Reply:
It is hard to assume what the "typical" juju deployment would look like. So I don't think it really matters. Spaces would help here, but I don't think we do that for api connections to controllers, so this is fine.